scripts/run_benchmark.py

```diff
@@ -300,6 +301,9 @@ def init_worker(env: Dict) -> NoReturn:
     )
 
     args = parser.parse_args()
+    if args.gpus[0] == -1:
+        print("Using All GPUs Detected.")
+        args.gpus = np.arange(torch.cuda.device_count()).tolist()
```

Comment:
If I'm not mistaken, `torch.cuda.device_count()` returns only the visible devices (please help to verify this). Say we have CUDA_VISIBLE_DEVICES="1,2" and the function should return 2, thus args.gpus becomes `[0, 1]`, which is not expected. I think the correct way to intepret '-1' should be either:

- Use all physical GPUs;
- Use all GPUs specified in CUDA_VISIBLE_DEVICES

which do you and @qiu69 think is the proper behavior?

Reply:
I would lean towards voting for `Use all GPUs specified in CUDA_VISIBLE_DEVICES`.

In a single-user setting, either approach can work well.

However, in a shared computing environment, it becomes crucial that the code respects the resource allocation and access regulations established by the system administrator.  By opting for `Use all GPUs specified in CUDA_VISIBLE_DEVICES`, the code will operate within the allocated GPU resources as defined by the `CUDA_VISIBLE_DEVICES` setting. This helps maintain fairness among different users sharing the computing resources and ensures that each user's allocated GPU subset is utilized as intended, without any one user overstepping their allocated resource boundaries.

Reply:
We've resolved all other comments. Since we agree on the solution, please fix this problem and the PR can be merged.

Reply:
I think maybe ‘Use all GPUs specified in CUDA_VISIBLE_DEVICES’ is more reasonable.